src/realizations/catchment/Bmi_C_Formulation.cpp

```diff
@@ -74,25 +74,32 @@ inline void Bmi_C_Formulation::get_forcing_data_ts_contributions(time_step_t t_d
                                                                  const std::vector<std::string> &param_units,
                                                                  std::vector<double> &summed_contributions)
 {
-    time_t model_ts_start_offset, model_ts_seconds_contained_in_forcing_ts;
-    bool increment_to_next_forcing_ts;
+    time_t model_ts_start_offset, model_ts_seconds_contained_in_forcing_ts, model_epoch_time_s;
     // Keep track of how much of the model ts delta has not yet had its contribution pulled from some forcing ts
     time_step_t contribution_seconds_remaining = t_delta;
 
-    // The sum of these first two is essentially the model time step's epoch start time in seconds
-    model_ts_start_offset = (time_t) (get_bmi_model()->convert_model_time_to_seconds(model_initial_time)) +
-                            get_bmi_model_start_time_forcing_offset_s() -
-                            forcing.get_time_epoch();
+    model_epoch_time_s = (time_t) (get_bmi_model()->convert_model_time_to_seconds(model_initial_time)) +
+                         get_bmi_model_start_time_forcing_offset_s();
+    // Make sure the forcings are properly primed to have the correct corresponding forcing ts data available
+    while (model_epoch_time_s > forcing.get_time_epoch() + forcing.get_time_step_size()) {
+        forcing.get_next_hourly_precipitation_meters_per_second();
+    }
+    // Though this is a problem at the moment, because we can't go through forcings in reverse order
+    if (model_epoch_time_s < forcing.get_time_epoch()) {
+        // TODO: think if there is a better way to address this
+        throw std::runtime_error("Cannot get contributions for model time step before current forcing time step");
```

Comment:
We need to get in the practice of identifying the class/function exceptions are thrown in, and relevant dynamic information, in this case the offending time step `model_epoc_time_s`.